apps/rush-lib/src/logic/PackageJsonUpdater.ts

```diff
@@ -293,82 +295,94 @@ export class PackageJsonUpdater {
       console.log(colors.green('Assigning "')
         + colors.cyan(initialSpec)
         + colors.green(`" for "${packageName}" because it matches what other projects are using in this repo.`));
-      return Promise.resolve(initialSpec);
+      return initialSpec;
     }
 
     if (this._rushConfiguration.ensureConsistentVersions && !initialSpec && implicitlyPinnedVersion) {
       console.log(`Assigning the version range "${colors.cyan(implicitlyPinnedVersion)}" for "${packageName}" because`
         + ` it is already used by other projects in this repo.`);
-      return Promise.resolve(implicitlyPinnedVersion);
+      return implicitlyPinnedVersion;
     }
 
     if (this._rushConfiguration.packageManager === 'yarn') {
       throw new Error('The Yarn package manager is not currently supported by the "rush add" command.');
     }
 
-    return installManager.ensureLocalPackageManager().then(() => {
-      let selectedVersion: string | undefined;
+    await installManager.ensureLocalPackageManager();
+    let selectedVersion: string | undefined;
 
-      if (initialSpec && initialSpec !== 'latest') {
-        console.log(colors.gray('Finding newest version that satisfies the selector: ') + initialSpec);
-        console.log();
-        console.log(`Querying registry for all versions of "${packageName}"...`);
-
-        const allVersions: string =
-          Utilities.executeCommandAndCaptureOutput(this._rushConfiguration.packageManagerToolFilename,
-            ['view', packageName, 'versions', '--json'],
-            this._rushConfiguration.commonTempFolder);
-
-        let versionList: Array<string> = JSON.parse(allVersions);
-        versionList = versionList.sort((a: string, b: string) => { return semver.gt(a, b) ? -1 : 1; });
+    if (initialSpec && initialSpec !== 'latest') {
+      console.log(colors.gray('Finding versions that satisfy the selector: ') + initialSpec);
+      console.log();
+      console.log(`Querying registry for all versions of "${packageName}"...`);
 
-        console.log(colors.gray(`Found ${versionList.length} available versions.`));
+      const allVersions: string =
+        Utilities.executeCommandAndCaptureOutput(this._rushConfiguration.packageManagerToolFilename,
+          ['view', packageName, 'versions', '--json'],
+          this._rushConfiguration.commonTempFolder);
 
```

Comment:
I don't know if prettier formats it this way, or if we have a style guideline, but IMO this would look _prettier_ formatted as: 
```
const allVersions: string = Utilities.executeCommandAndCaptureOutput(
  this._rushConfiguration.packageManagerToolFilename,
  ['view', packageName, 'versions', '--json'],
  this._rushConfiguration.commonTempFolder
);

Reply:
Agreed.